### PR TITLE
Add generic codec for CCIP plugin

### PR DESCRIPTION
**Pull request by huangzhen1997**
https://smartcontract-it.atlassian.net/browse/NONEVM-935 

This is part of the change for supporting solana plugin codec. The plugin component will use generic codec for encode and decode commit and exec report. This PR adds a new commitReportCodec and execReportCodec using the evm codec. Once Solana codec is fully implemented, we will add the solana implementation under ccipsolana

### CODE REVIEW

File: core/capabilities/ccip/ccipevm/commitcodecv2.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+)
+
+const commitReportABI = `[{"components":[{"name":"chainSel","type":"uint64","internalType":"uint64"},{"name":"onRampAddress","type":"bytes","internalType":"bytes"},{"name":"seqNumsRange","type":"uint64[2]","internalType":"uint64[2]"},{"name":"merkleRoot","type":"bytes32","internalType":"bytes32"}],"name":"merkleRoots","type":"tuple[]","internalType":"struct MerkleRootChain[]"},{"components":[{"components":[{"name":"tokenID","type":"string","internalType":"string"},{"name":"price","type":"uint256","internalType":"uint256"}],"name":"tokenPriceUpdates","type":"tuple[]","internalType":"struct TokenPrice[]"},{"components":[{"name":"chainSel","type":"uint64","internalType":"uint64"},{"name":"gasPrice","type":"uint256","internalType":"uint256"}],"name":"gasPriceUpdates","type":"tuple[]","internalType":"struct GasPriceChain[]"}],"name":"priceUpdates","type":"tuple","internalType":"struct PriceUpdates"},{"components":[{"name":"r","type":"bytes32","internalType":"bytes32"},{"name":"s","type":"bytes32","internalType":"bytes32"}],"name":"rmnSignatures","type":"tuple[]"}]`
```
Comment (winder) [inline core/capabilities/ccip/ccipevm/commitcodecv2.go:16]: Would be generated from the contracts eventually?

File: core/capabilities/ccip/ccipevm/commitcodecv2.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+func (c *CommitPluginCodecV2) Encode(ctx context.Context, report cciptypes.CommitPluginReport) ([]byte, error) {
+	if err := validateReport(report); err != nil {
+		return nil, err
+	}
+
+	cd, err := codec.NewCodec(commitCodecConfig)
+	if err != nil {
+		return nil, err
+	}
```
Comment (winder) [inline core/capabilities/ccip/ccipevm/commitcodecv2.go]: initialize in constructor

File: core/capabilities/ccip/ccipevm/commitcodecv2.go (hunk 1 above)
Comment (huangzhen1997) [reply core/capabilities/ccip/ccipevm/commitcodecv2.go:16]: Yes, that would be ideal. The existing generated [ABI](https://github.com/smartcontractkit/chainlink/blob/86d5f9993ef208ef0a310a781b198563f4ae7d55/core/gethwrappers/ccip/generated/report_codec/report_codec.go#L101) seems match with the format from generated [MerkleRoot](https://github.com/smartcontractkit/chainlink/blob/86d5f9993ef208ef0a310a781b198563f4ae7d55/core/gethwrappers/ccip/generated/ccip_encoding_utils/ccip_encoding_utils.go#L61), not with the original report [format](https://github.com/smartcontractkit/chainlink-ccip/blob/8b4b62f86e5716b17253d7529e0e688496deefdc/pkg/types/ccipocr3/plugin_commit_types.go#L43). See discussion from slack [thread](https://chainlink-core.slack.com/archives/C0779R3AH8R/p1733178788746109?thread_ts=1733177120.342399&cid=C0779R3AH8R)

**Comment by huangzhen1997** [comment]
Skip EVM generic codec for CCIP plugin at the moment https://chainlink-core.slack.com/archives/C0779R3AH8R/p1733513465890859?thread_ts=1733177120.342399&cid=C0779R3AH8R

**Pull request closed without merging**